Automattic/woocommerce-payments
# Fix checkout broken when user has bad account details store in meta.

Fixes #818

#### Changes proposed in this Pull Request

* Handle an Exception that should not break the checkout.
* Add tests.
* Fix error log message with duplicate wording.
#### Testing instructions



* For the first part you should be on the Master branch.
* First lets break things. Run these 2 commands: 
1.  `npm run wp user meta delete 1 wp__wcpay_customer_id `
2. `npm run wp user meta add 1 wp__wcpay_customer_id cus_1234isold`

* Next load up your store, add something to cart, and then visit the checkout page. 
* You should see that the credit card field no longer loads.

<img width="471" alt="Screenshot 2020-09-11 at 14 58 00" src="https://user-images.githubusercontent.com/1713474/92928341-3b1b8e00-f43f-11ea-8536-3a53ce66dae9.png">

* Now, check out this PR branch and see the card field show up again.
-------------------
